diff --git a/node_modules/axios/lib/adapters/fetch.js b/node_modules/axios/lib/adapters/fetch.js
index f997f1d..e5550d1 100644
--- a/node_modules/axios/lib/adapters/fetch.js
+++ b/node_modules/axios/lib/adapters/fetch.js
@@ -151,7 +151,10 @@ export default isFetchSupported && (async (config) => {
       }
     }
 
-    if (!utils.isString(withCredentials)) {
+    // credential field is set by default (https://github.com/axios/axios/pull/6505)
+    // which is causing issue when running cloudflare worker (https://github.com/cloudflare/workers-sdk/issues/2514)
+    const isCredentialsSupported = "credentials" in Request.prototype;
+    if (isCredentialsSupported && !utils.isString(withCredentials)) {
       withCredentials = withCredentials ? 'include' : 'omit';
     }
 
